components/ca-form.js

```diff
@@ -0,0 +1,1520 @@
+        }
+
+        /** @return {string} gets thew reset value */
+        get reset() {
+            return this.getAttribute('reset');
+        }
+
+        /** @param {string} value, gets the reset value */
+        set reset(value) {
+            return this.setAttribute('reset', value);
+        }
+
+        /**
+         * @description Loads the schema
+         * @returns {undefined} nothing.
+        */
+        loadSchema() {
+
+            const schemaUrl = this.src;
+
+            return componentSupport.request({ url: schemaUrl, dataType: 'json' }).then(data => {
+                this._schema = [data];
+                this.render();
+
+                // fire on schema loaded event
+                this.onschemaloaded(this, data.title || '', data);
+
+                // apply session stored form data if it exists
+                if (this.persist && window.sessionStorage && sessionStorage[this.src]) {
+
+                    const formData = sessionStorage[this.src] || '';
+
+                    if (formData !== '') {
+                        this.populate(JSON.parse(formData));
+                    }
+                }
+            }).catch(err => {
+                // eslint-disable-next-line no-alert
+                alert(`Schema not found! ${err}`);
+            });
+        }
+
+        /**
+         * @description Builds the HTML form based on the value of the assigned JSON .schema object
+         * @returns {magic} tricks.
+        */
+        render() {
+
+            // build table headers from schema
+            if (this._schema) {
+
+                // erase current form
+                clearElement(this.form);
+
+                // grab a handle to elements for rendering
+                const schema = this.selectedSchema;
+                let itemIndex = 1;
+                let group = '';
+                const schemaTitle = this._title || schema.title || '';
+                const schemaDesc = this._description || (Array.isArray(schema.description) ? schema.description.join('') : schema.description || '');
+
+                // insert title if it has one
+                if (schemaTitle !== '') {
+                    createElementLegacy(this.form, 'h2', { 'data-schema-title': schemaTitle }, null, schemaTitle);
+                }
+
+                if (schemaDesc !== '') {
+                    createElementLegacy(this.form, 'div', { class: 'ca-form-desc', 'data-schema-description': schemaDesc }, null, schemaDesc);
+                }
+
+                // add the selected schema to the form element
+                this.form.setAttribute('data-schema-type', schema.title);
+
+                // add schema option if more than one exist
+                if (this.schema.length > 1) {
+
+                    // create a form label container (acts as a row)
+                    const lbl = createElementLegacy(null, 'label', { for: 'schemaType', 'data-key': 'schemaType' });
```

Comment:
shorthand like this is not a good idea imo, change to label or rowLabel or something more descriptive